src/Symfony/Component/BrowserKit/HttpBrowser.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\BrowserKit;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\HttpKernel\HttpCache\HttpCache;
+use Symfony\Component\HttpKernel\HttpCache\Store;
+use Symfony\Component\HttpKernel\HttpKernelBrowser;
+use Symfony\Component\HttpKernel\RealHttpKernel;
+use Symfony\Component\Mime\Part\AbstractPart;
+use Symfony\Component\Mime\Part\Multipart\FormDataPart;
+use Symfony\Component\Mime\Part\TextPart;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * An implementation of a browser using the HttpClient component
+ * to make real HTTP requests.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+final class HttpBrowser extends AbstractBrowser
+{
+    private $client;
+    private $store;
+    private $logger;
+    private $cache;
+    private $httpKernelBrowser;
+
+    public function __construct(HttpClientInterface $client = null, Store $store = null, LoggerInterface $logger = null)
+    {
+        if (!class_exists(HttpClient::class)) {
+            throw new \LogicException(sprintf('You cannot use "%s" as the HttpClient component is not installed. Try running "composer require symfony/http-client".', __CLASS__));
+        }
+
+        $this->client = $client ?? HttpClient::create();
+        $this->store = $store;
+        $this->logger = $logger ?? new NullLogger();
+
+        parent::__construct();
+    }
+
+    protected function doRequest($request)
+    {
+        if ($this->store) {
+            return $this->doCachedHttpRequest($request);
```

Comment:
Are there plans to have a `CachingHttpClient`?

Reply:
plans no, but you know how Symfony works: contributions fuel it forward ;)